library/catalog/models.py

```diff
@@ -45,6 +46,44 @@ class MongoUser(Document):
     rated_books = DictField(default={})
     preference = EmbeddedDocumentField(Preference.__name__, default=Preference())
 
+    def make_recommended_books_list(self, books_limit=50):
+        if self.wishlist:
+            wishlist_books = [Book.objects(id=book_id).first() for book_id in self.wishlist]
+            wishlist_books.sort(key=lambda book: book.statistic.rating, reverse=True)
+
+            books_by_author_and_genre = []
+            books_by_author = []
+            books_by_genre = []
+
+            for book in wishlist_books:
+                books_by_author_and_genre += list(Book.objects.filter(
+                    Q(author_id=book.author_id) & Q(genres__icontains=book.genres)).order_by('-statistic__rating'))
+                books_by_author += list(Book.objects.filter(author_id=book.author_id).order_by('-statistic__rating'))
+                books_by_genre += list(
+                    Book.objects.filter(genres__icontains=book.genres).order_by('-statistic__rating'))
+
+            recommendations_list = books_by_author_and_genre + books_by_author + books_by_genre
```

Comment:
If we use set for `books_by_author` and `books_by_genre ` we may avoid checks if value is unique.

Reply:
I check if value is unique in 'for-while' loop

Reply:
Reverse function need here for delete from the end of collection and back collection to starting state, order of the books is very important